frontend/src/pages/exploreApplication/EnableModal.tsx

```diff
@@ -54,15 +54,16 @@ const EnableModal: React.FC<EnableModalProps> = ({ selectedApp, shown, onClose }
   React.useEffect(() => {
     if (validationInProgress && validationStatus === EnableApplicationStatus.SUCCESS) {
       setValidationInProgress(false);
-      selectedApp.spec.isEnabled = true;
-      selectedApp.spec.shownOnEnabledPage = true;
+      const updatedSelectedApp = { ...selectedApp };
+      updatedSelectedApp.spec.isEnabled = true;
+      updatedSelectedApp.spec.shownOnEnabledPage = true;
```

Comment:
This changes the current logic.
The modification in place is expected.

However this approach is completely wrong. We should not be modifying the object in place but rather notifying the called so whomever owns the object can modify it and / or react to the situation.

The simplest solution today would be to disable the rule here and add a `TODO` comment.